lib/livebook/utils.ex

```diff
@@ -303,4 +303,50 @@ defmodule Livebook.Utils do
     data = Base.encode64(content)
     "data:#{mime};base64,#{data}"
   end
+
+  @doc """
+  Splits the given string at the last occurrence of `pattern`.
+
+  ## Examples
+
+      iex> Livebook.Utils.split_at_last_occurrence("1,2,3", ",")
+      {:ok, "1,2", "3"}
+
+      iex> Livebook.Utils.split_at_last_occurrence("123", ",")
+      :error
+  """
+  @spec split_at_last_occurrence(String.t(), String.pattern()) ::
+          {:ok, left :: String.t(), right :: String.t()} | :error
+  def split_at_last_occurrence(string, pattern) when is_binary(string) do
+    case :binary.matches(string, pattern) do
+      [] ->
+        :error
+
+      parts ->
+        {start, _} = List.last(parts)
+        size = byte_size(string)
+        {:ok, binary_part(string, 0, start), binary_part(string, start + 1, size - start - 1)}
+    end
+  end
+
+  @doc ~S"""
+  Finds CR characters and removes leading text in the same line.
+
+  ## Examples
+
+      iex> Livebook.Utils.apply_rewind("Hola\nHmm\rHey")
+      "Hola\nHey"
+
+      iex> Livebook.Utils.apply_rewind("\rHey")
+      "Hey"
+  """
+  @spec apply_rewind(String.t()) :: String.t()
+  def apply_rewind(string) when is_binary(string) do
+    string
+    |> String.split("\n")
+    |> Enum.map(fn line ->
+      String.replace(line, ~r/^.*\r([^\r].*)$/, "\\1")
```

Comment:
We could use split last here too, no?

Reply:
Not really, we may get new text like this:

```
line 1
line 2 \r line 3 \r line 4
line 5 \r line 6 \r line 7
```

This doesn't look like something we would get on a single print, but the group leader buffers output, so we may actually get this and we want to make sure we strip it to the minimal form. Does that make sense?